package tera;

import "table_meta.proto";

enum QuotaOperationType {
    kQuotaWriteReqs = 1;
    kQuotaWriteBytes = 2;
    kQuotaReadReqs = 3;
    kQuotaReadBytes = 4;
    kQuotaScanReqs = 5;
    kQuotaScanBytes = 6;
}

message QuotaInfo {
    required QuotaOperationType type = 1;
    required int64 limit = 2;
    // default 1s
    required int64 period = 3 [default = 1];
}

message TableQuota {
    enum TableQuotaType {
        kSetQuota = 1;
        kDelQuota = 2;      // when type == kDelQuota, doesn't need quota_infos.
    }
    required string table_name = 1;
    repeated QuotaInfo quota_infos = 2;
    required TableQuotaType type = 3 [default = kSetQuota];
}

message TsQuota {
    required string ts_addr = 1;
    repeated TableQuota table_quotas = 2;
}
